crypto/keys/multisig/multisig.go

```diff
@@ -40,23 +42,27 @@ func (m *LegacyAminoPubKey) Bytes() []byte {
 	return AminoCdc.MustMarshalBinaryBare(m)
 }
 
-// VerifyMultisignature implements the multisigtypes.PubKey VerifyMultisignature method
+// VerifyMultisignature implements the multisigtypes.PubKey VerifyMultisignature method.
+// The signatures must be added in an order corresponding to the public keys order in
+// LegacyAminoPubKey. It's OK to have multiple same keys in the multisig - it will increase
```

Comment:
Why not block this, and force weighted multisigs to use a different type?

Reply:
Are you suggesting we should not allow duplicate sigs to be constructed?

Reply:
Yes, I don't think duplicate pubkeys / sigs is a good thing to have

Reply:
That was my original intention. We already had a lengthy discussion for this. In the end we decided to not change the behavior because:
* some people may want it
* It may already be used (however we could only check on the constructor)
* the whole `LegacyAmintoMultisig` is deprecated and we want to change it further.

Reply:
Here is the original implementation from this PR, which I removed later: https://github.com/cosmos/cosmos-sdk/pull/8600/commits/bae3332a4ec6f640d547558bb048d03172993214

Reply:
> the whole LegacyAmintoMultisig is deprecated and we want to change it further.

That is the main reason we don't want to carry out too many changes. A multisig replacement is already in the works

Reply:
Exactly - that was one of the reasons to remove the uniqueness check.

Reply:
BTW, @alessio , which replacement do you have in mind?

Reply:
The Group Module

Reply:
I see, that makes sense.